From 7aea24c6ac102f8e294ed290cf470e9733cc4d0b Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 19 Feb 2014 14:25:25 +0800
Subject: [PATCH 289/602] ENGR00299781 - testExternalStorageGifts, xxxNone,
 xxxRead fail

kk4.4 support multi-user, external storage for apk is under
"/storage/emulated/uid". For example, if root, then under
"/storage/emulated/0".

sdcard process need add -l option to support DERIVE_LEGACY.
Below is comments from system/core/sdcard/sdcard.c
/* Legacy behavior used to support internal multiuser layout which
 * places user_id at the top directory level, with the actual roots
 * just below that. Shared OBB path is also at top level. */

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 imx6/etc/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 1a1b812..055b0bd 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -722,7 +722,7 @@ service wifi_mac /system/bin/sh /system/etc/check_wifi_mac.sh
 
 # virtual sdcard daemon running as media_rw (1023)
 # default 4 threads have better performance to 2 threads.
-service sdcard /system/bin/sdcard -t4 /data/media /mnt/shell/emulated 1023 1023
+    service sdcard /system/bin/sdcard -l -t4 /data/media /mnt/shell/emulated 1023 1023
     class late_start
 
 # Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
-- 
1.8.0

